RELEASE.md

```diff
@@ -0,0 +1,52 @@
+# Release Process for `connect-go`
+
+## Preparing for Release
+
+1. **Update Version Number**
+   - In `connect.go`, modify the `Version` constant to match the target release according to [semver](https://semver.org/) standards. Remove the `-dev` suffix.
+     ```go
+     const Version = "X.Y.Z"
+     ```
+
+2. **Version Requirements (Optional)**
```

Comment:
This is not an optional step. You are required to check to see if the new release requires an updated version of connect-go.

Reply:
This step should always be done in the PR that makes the change. As an example see the new schema addition which set the requirement on v1.13.0: https://github.com/connectrpc/connect-go/pull/629/files#diff-82f6a0eb10c47bc6590f594adc997b5a12de370dfd735a05c44cf8ff6fbf7a12R44
Appropriate to remove this step?

Reply:
I've removed this step. It should never be done in the release process but it's not well documented. Would a section in this file help?

Reply:
Looking through the history it's not clear what the correct procedure is. For adding GET support https://github.com/connectrpc/connect-go/pull/478 the version tag was added for v1.6.0 after the v1.6.0 release. This was then corrected in the version release https://github.com/connectrpc/connect-go/pull/496 for v1.7.0. I do think the version restriction should be done in the PR landing the change as it would be easy to miss in a release. The current docs are:

3. **If the generated code has begun using a new API from the main package**:
    1. Define a new `IsAtLeastVersion_X_Y_Z` constant in `connect.go`.
    2. Change `cmd/protoc-gen-connect-go/main.go` to use the new constant.
    3. Commit with the message “Prepare for vX.Y.Z”, substituting the actual version we’re releasing.

The **if** suggests it is optional and the substep 3.3 is not dependant on changing the `main.go` as we must create a PR with the connect.go changes that has this same title.

Reply:
Agree that best practice is for this version constant to land in the original PR and the existing docs are a bit misleading. It does seem valuable for the releaser to double check to see if any new constants have been added since the last release and to ensure they are named correctly. Basically what are the instructions we should add here that would have guided a releaser to detect the problem that was detected in [#496](https://github.com/connectrpc/connect-go/pull/496) (aka wrong version number used).

Reply:
Added step 3. to check for the version restrictions if there have been changes to protoc-gen-connect-go.